packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -58,41 +63,76 @@ class ${class_name}:
         Args:
             discovery_client: An optional discovery client.
 
+            pin_map_client: An optional pin map client.
+
             grpc_channel: An optional gRPC channel targeting a measurement service.
 
             grpc_channel_pool: An optional gRPC channel pool.
         """
-        self._initialization_lock = threading.Lock()
+        self._initialization_lock = threading.RLock()
         self._service_class = "${service_class}"
         self._grpc_channel_pool = grpc_channel_pool
         self._discovery_client = discovery_client
+        self._pin_map_client = pin_map_client
         self._stub: Optional[v2_measurement_service_pb2_grpc.MeasurementServiceStub] = None
         self._configuration_metadata = ${configuration_metadata}
         self._output_metadata = ${output_metadata}
         if grpc_channel is not None:
             self._stub = v2_measurement_service_pb2_grpc.MeasurementServiceStub(grpc_channel)
         self._create_file_descriptor()
-        
+        self._pin_map_id = ""
+        self._sites = [0]
+
+    @property
+    def sites(self) -> Tuple[int, ...]:
+        """Sites for which the measurement is being executed."""
+        return tuple(self._sites)
+    
+    def set_sites(self, val):
```

Comment:
Please add type-hinting for the parameter & rename the parameter to sites instead of val

Reply:
Not applicable as we removed sites property.